### PR TITLE
MWPW-136203 | Fixing empty blog post section to load current locale query-index json

### PR DESCRIPTION
**Pull request by apganapa**
Fixing empty blog post section to load current locale query-index json

While testing today, noticed that empty blog post sections are added on pages just to load the current query-index.json file so that the tags can be loaded based on the link comparison and fetching data for each tags.
This PR fixes it by adding original logic and condition to check if the locale is processed or not.

Resolves: MWPW-136203

Test URLs:

- Before: https://main--express--adobecom.hlx.page/drafts/apganapa/blog-fr?lighthouse=on
- After: https://blog-fix--express--adobecom.hlx.page/drafts/apganapa/blog-fr?lighthouse=on

- Before: https://main--express--adobecom.hlx.page/uk/express/learn/blog/introducing-creative-cloud-express?lighthouse=on
- After: https://blog-fix--express--adobecom.hlx.page/uk/express/learn/blog/introducing-creative-cloud-express?lighthouse=on

### CODE REVIEW

File: express/blocks/blog-posts/blog-posts.js [hunk 1]
```diff
@@ -21,15 +21,24 @@ import {
 } from '../../scripts/scripts.js';
 
 async function fetchBlogIndex(config) {
-  let prefix = getLocale(window.location);
-  if (prefix === 'express' || prefix === 'drafts' || prefix === 'documentation' || prefix === 'us') prefix = '';
+  const prefix = getLocale(window.location);
+  let currentLocalePrefix;
+  let currentLocaleProcessed = false;
+  if (prefix === 'express' || prefix === 'drafts' || prefix === 'documentation' || prefix === 'us') {
```
Comment (qiyundai) [inline express/blocks/blog-posts/blog-posts.js:27]: I actually just realized that the getLocale() should never give you the first three conditions. The regex in getLocale() tests against only 2 letters string. You should be able to save yourself some bytes here.

Comment (apganapa) [reply express/blocks/blog-posts/blog-posts.js]: @qiyundai done, took care of it and tested, you are right.

File: express/blocks/blog-posts/blog-posts.js [hunk 2]
```diff
@@ -42,7 +37,7 @@ async function fetchBlogIndex(config) {
       if (localePrefix === 'us') {
         localePrefix = '';
       }
-      if (linkLocales.indexOf(localePrefix) === -1 && !(localePrefix === 'drafts' || localePrefix === 'documentation')) {
+      if (linkLocales.indexOf(localePrefix) === -1) {
```
Comment (qiyundai) [inline express/blocks/blog-posts/blog-posts.js:40]: Sorry I didn't catch this the first time -- could we use `!linkLocales.includes(localePrefix)` ?

Comment (apganapa) [reply express/blocks/blog-posts/blog-posts.js]: done

**Review by qiyundai — APPROVED** [review]

**Pull request merged**